exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetReaderUtility.java

```diff
@@ -45,4 +53,34 @@ public static int getIntFromLEBytes(byte[] input, int start) {
     }
     return out;
   }
+
+  /**
+   * Utilities for converting from parquet INT96 binary (impala, hive timestamp)
+   * to date time value. This utilizes the Joda library.
+   */
+  public static class NanoTimeUtils {
+
+    public static final long NANOS_PER_DAY = TimeUnit.DAYS.toNanos(1);
+    public static final long NANOS_PER_HOUR = TimeUnit.HOURS.toNanos(1);
+    public static final long NANOS_PER_MINUTE = TimeUnit.MINUTES.toNanos(1);
+    public static final long NANOS_PER_SECOND = TimeUnit.SECONDS.toNanos(1);
+    public static final long NANOS_PER_MILLISECOND =  TimeUnit.MILLISECONDS.toNanos(1);
+
+  /**
+   * @param binaryTimeStampValue
+   *          hive, impala timestamp values with nanoseconds precision
+   *          are stored in parquet Binary as INT96
+   *
+   * @return  the number of milliseconds since January 1, 1970, 00:00:00 GMT
+   *          represented by @param binaryTimeStampValue .
+   */
+    public static long getDateTimeValueFromBinary(Binary binaryTimeStampValue) {
+      NanoTime nt = NanoTime.fromBinary(binaryTimeStampValue);
+      int julianDay = nt.getJulianDay();
+      long nanosOfDay = nt.getTimeOfDayNanos();
+      return DateTimeUtils.fromJulianDay(julianDay-0.5d) + nanosOfDay/NANOS_PER_MILLISECOND;
```

Comment:
Sorry for the late reply. For some reason, I did not see these comments till now. 
About 1) Yes, you are correct. I just want the comments in ConvertFromImpalaTimestamp to be removed.